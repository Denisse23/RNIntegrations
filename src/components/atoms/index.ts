export { default as CustomText } from './CustomText/CustomText';
export { default as Button } from './Button/Button';
export { default as CustomInput } from './CustomInput/CustomInput';
export { type CustomInputProps } from './CustomInput/CustomInput';
