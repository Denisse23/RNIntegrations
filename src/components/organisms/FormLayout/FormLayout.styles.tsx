import styled from '@emotion/native';

export const StyledTouchableWithoutFeedback = styled.TouchableWithoutFeedback(
  () => ({
    flex: 1,
  }),
);
