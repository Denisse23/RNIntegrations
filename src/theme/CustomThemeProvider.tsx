import React from 'react';
import { useColorScheme } from 'react-native';
import { ThemeProvider } from '@emotion/react';
import { ColorScheme, DarkTheme, LightTheme } from './Themes';

const CustomThemeProvider = ({ children }: { children: React.ReactNode }) => {
  const scheme = useColorScheme();

  return (
    <ThemeProvider theme={scheme === ColorScheme.DARK ? DarkTheme : LightTheme}>
      {children}
    </ThemeProvider>
  );
};

export default CustomThemeProvider;
