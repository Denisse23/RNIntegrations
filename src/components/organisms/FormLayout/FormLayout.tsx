import React from 'react';
import { Keyboard, ViewProps } from 'react-native';
import { Layout } from '@components/organisms';
import { StyledTouchableWithoutFeedback } from './FormLayout.styles';

export interface FormLayoutProps {
  testID?: ViewProps['testID'];
  children?: React.ReactNode;
}

const FormLayout = ({ testID, children }: FormLayoutProps) => {
  return (
    <Layout testID={testID}>
      <StyledTouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
        {children}
      </StyledTouchableWithoutFeedback>
    </Layout>
  );
};

export default FormLayout;
