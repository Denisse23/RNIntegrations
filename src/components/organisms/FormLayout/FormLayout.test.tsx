import React from 'react';
import renderer from 'react-test-renderer';
import FormLayout from './FormLayout';
import { CustomText } from '@components/atoms';
import CustomThemeProvider from '@theme/CustomThemeProvider';

describe('FormLayout', () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(
      <CustomThemeProvider>
        <FormLayout>
          <CustomText>Test</CustomText>
        </FormLayout>
      </CustomThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
