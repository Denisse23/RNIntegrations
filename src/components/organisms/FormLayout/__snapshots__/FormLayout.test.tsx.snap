// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLayout should match the snapshot 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Array [
              Object {
                "padding": 37.5,
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "color": "#353434",
                "fontSize": 28.125,
                "fontWeight": "400",
              },
              undefined,
            ]
          }
        >
          Test
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
